connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -520,6 +554,9 @@ protected void handleLostAssignments(ConnectorsAndTasks lostAssignments,
                 newSubmissions.tasks().addAll(lostAssignments.tasks());
             }
             resetDelay();
+            // Resetting the flag as now we can permit successive revoking rebalances.
+            // since we have gone through the full rebalance delay
+            revokedInPrevious = false;
```

Comment:
I did think about it.. Problem is if we set the counter to 0 here every time then the exponential backoff would only go upto initial interval after every successive revoking rebalance. The other reason I didn't reset this is that since max interval is maxDelay, this value can be incremented w/o introducing delays > maxDelay. One thing to consider could be can this value overflow if we let it increase. Let me know what you think about it.

Reply:
OK, I got your point. The result will be something like this, right?
1. rebalance: [revokedInPrevious = false, numSuccessive = 0] -> revoke
2. rebalance: [revokedInPrevious = true, numSuccessive = 1] -> delay n 
3. after n (clear revokedInPrevious)
4. rebalance: [revokedInPrevious = false, numSuccessive = 1] -> revoke
5. rebalance: [revokedInPrevious = true, numSuccessive = 1] -> reset numSuccessive or delay

Reply:
Yeah so point #5, with the approach I have would set numSuccessive = 2 and calculate backoff delay based on that. If I reset numSuccessive to 0 in #3 from above, then at 5, it would be back to `rebalance: [revokedInPrevious = true, numSuccessive = 1] ` and delay would be equal to the delay we had at #2 above.

Reply:
Yes, make sense to me, to not reseting numSuccessiveRevokingRebalances here. Thanks.